src/Microsoft.AspNetCore.Mvc.Core/Formatters/StringOutputFormatter.cs

```diff
@@ -29,18 +29,17 @@ public override bool CanWriteResult(OutputFormatterCanWriteContext context)
                 throw new ArgumentNullException(nameof(context));
             }
 
-            // Ignore the passed in content type, if the object is string
-            // always return it as a text/plain format.
             if (context.ObjectType == typeof(string) || context.Object is string)
             {
-                if (!context.ContentType.HasValue)
+                // Call into base to check if the current request's content type is a supported media type.
+                var canWriteResult = base.CanWriteResult(context);
```

Comment:
I agree only the XML formatters override `OutputFormatter.CanWriteType()` (which isn't passed the context or the model) and none of the other `CanWriteResult()` overrides call the `base` method. But, there's no need to expand the `OutputFormatter` API for this one use case. 

FYI however `OutputFormatter` exposes a `GetSupportedContentTypes()` which does an inefficient version of what you want. Pity it always returns a new `List<string>` and definitely shouldn't be used here.